src/Lykke.Service.ConfirmationCodes.Services/ConfirmationCodesService.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Lykke.Service.ClientAccount.Client;
+using Lykke.Service.ConfirmationCodes.Core.Entities;
+using Lykke.Service.ConfirmationCodes.Core.Heplers;
+using Lykke.Service.ConfirmationCodes.Core.Messages;
+using Lykke.Service.ConfirmationCodes.Core.Repositories;
+using Lykke.Service.ConfirmationCodes.Core.Services;
+
+namespace Lykke.Service.ConfirmationCodes.Services
+{
+    internal class ConfirmationCodesService : IConfirmationCodesService
+    {
+        private readonly ISmsVerificationCodeRepository _smsVerificationCodeRepository;
+        private readonly ISmsRequestProducer _smsRequestProducer;        
+        private readonly ISupportToolsSettings _supportToolsSettings;
+        private readonly ICallTimeLimitsRepository _callTimeLimitsRepository;
+        private readonly IClientAccountClient _clientAccountService;
+        private readonly IDeploymentSettings _deploymentSettings;
+        
+        private const string GetMethodName = "ConfirmationCodesService.RequestSmsCode";
+        private readonly TimeSpan _repeatCallsTimeSpan = TimeSpan.FromMinutes(5);
```

Comment:
Maybe it is better use settings for this values?

Reply:
Out of scope - will be refactored after